### ISSUE TITLE
Add support for GraalVM native images

### ISSUE THREAD

**Issue by gtskaushik** [vladmihalcea/hibernate-types#343]
Hi Team
Thanks for your work on HibernateTypes. When we are trying to build a native image for Spring Boot, the build succeeds but while running, it fails with the following error:
`
Native reflection configuration for org.hibernate.annotations.common.reflection.java.JavaXAnnotatedElement.getJavaType() is missing.

Action:

Native configuration for a method accessed reflectively is likely missing
`

Following are the type hints added
`
@NativeHint(types = {
				@TypeHint(types = {
						org.springframework.jdbc.datasource.SimpleDriverDataSource.class,
						com.vladmihalcea.hibernate.type.basic.PostgreSQLEnumType.class,
                        com.vladmihalcea.hibernate.type.json.JsonBinaryType.class
				})
`

Spring Boot version - 2.5.4
Spring Native version - 0.10.3-SNAPSHOT

**Comment by edudar** [comment]
I've spent some time trying to track this down as I faced the same issue with JSON types. So far, I think the culprit is [this](https://github.com/vladmihalcea/hibernate-types/blob/master/hibernate-types-55/src/main/java/com/vladmihalcea/hibernate/type/json/internal/JsonTypeDescriptor.java#L69) reflection call to get `javaType`. I don't know hibernate-types good enough and not sure why it needs to be a reflection call here, but doing `((JavaXMember) xProperty).getJavaType()`, after all, we just checked type that it's `JavaXMember`, allowed me to move past Hibernate issues.

@vladmihalcea I can make a PR of this sounds like a proper fix.
